dvc/remote/local.py

```diff
@@ -346,7 +346,7 @@ def checkout(self, path_info, checksum_info, force=False):
     def _already_cached(self, path):
         current_md5 = self.state.update(path)
 
-        return not self.changed_cache(current_md5)
+        return current_md5 and self.changed_cache(current_md5)
```

Comment:
@efiop , what do you think we can do to mitigate this kind of errors?

Currently, we are the only users of `dvc`'s library (i.e. The _public_ methods that we expose), but if we are planning to support a `dvc` library where users can interact with their dvc projects using Python (_a la_  [gitpython](https://gitpython.readthedocs.io/en/stable/)), we should think about every _public_ method and property as possible endpoints to receive user input, and we should start validating the arguments.

I don't have any clear answer right now, but I think is worth to talk about on Friday :slightly_smiling_face:

Reply:
I agree, dvc library lacks proper validation. For now we should simply add a test for it, to make sure that it doesn't happen in the future. Something like
```
#!/bin/bash

set -e
set -x

rm -rf myrepo
mkdir myrepo
cd myrepo

git init
dvc init

mkdir dir
echo foo > dir/foo

dvc add dir
rm -rf dir
dvc checkout
```
^^^NOT TESTED

Please make sure the test fails without this patch, so we don't get a false positive.